### PR TITLE
feat!: support new `instrumentation` paradigm

### PR DESCRIPTION
**Pull request by RogerHYang**
For more info, see https://docs.llamaindex.ai/en/stable/module_guides/observability/instrumentation/

To use the legacy callback handler instead, set `use_legacy_callback_handler=True` as shown below.
```python
LlamaIndexInstrumentor().instrument(use_legacy_callback_handler=True)
```

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-llama-index/examples/chat_engine_stream_chat.py [hunk 1]
```diff
@@ -26,6 +26,6 @@
 Settings.llm = OpenAI(model="gpt-3.5-turbo")
 
 if __name__ == "__main__":
-    query_engine = index.as_query_engine(streaming=True, similarity_top_k=1)
-    response_stream = query_engine.query("What did the author do growing up?")
+    query_engine = index.as_chat_engine()
+    response_stream = query_engine.stream_chat("What did the author do growing up?")
```
Comment (axiomofjoy) [inline python/instrumentation/openinference-instrumentation-llama-index/examples/chat_engine_stream_chat.py:30]:
```suggestion
    chat_engine = index.as_chat_engine()
    response_stream = chat_engine.stream_chat("What did the author do growing up?")
```

Is this still a query engine? Maybe we should rename the variable?

File: python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/__init__.py [hunk 2]
```diff
@@ -30,40 +27,28 @@ def instrumentation_dependencies(self) -> Collection[str]:
...
+            if hasattr(llama_index.core, "global_handler"):
+                print("Using legacy callback handler.")
             else:
-                print(
-                    f"`use_experimental_instrumentation` feature flag is set. But "
-                    f"the version of `llama-index-core` is not "
-                    f">={'.'.join(map(str, _ELIGIBLE_VERSION_FOR_NEW_INSTRUMENTATION))}, "
-                    f"so the flag is ignored."
-                )
+                print("Legacy callback handler is not available.")
```
Comment (axiomofjoy) [inline python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/__init__.py:37]:
```suggestion
                print("Legacy callback handler is not available. Using new callback handler.")
```

Suggestion: A slightly more verbose message would be helpful here.

**Review by axiomofjoy — APPROVED** [review]

File: python/instrumentation/openinference-instrumentation-llama-index/examples/chat_engine_stream_chat.py (hunk 1 above)
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-llama-index/examples/chat_engine_stream_chat.py:30]: you're right, my oversight. thx!

File: python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/__init__.py (hunk 2 above)
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/__init__.py:37]: ok

**Review by axiomofjoy — APPROVED** [review]

**Pull request merged**